app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -70,4 +73,21 @@ def not_found(options = nil)
 
     false
   end
+
+  def registration_url
+    uri = if params[:url].present?
+            URI.join(params[:url], '/register')
+          else
+            URI(register_url)
+          end
+
+    # Without protocol (but with port) uri.host returns nil
+    # URI('example.com:99').host => nil
+    if uri.to_s.starts_with?('http://', 'https://')
```

Comment:
Does this work? Is `starts_with?` a Rails extension to String? Does it make sense to use the explicit properties on the URI object?
```suggestion
    if uri.scheme && uri.host
```